core/src/value/primitive.rs

```diff
@@ -604,6 +604,31 @@ impl PrimitiveValue {
         }
     }
 
+    /// Getter method to provide a clean version of a string or multiple strings,
+    /// removing unwanted whitespaces
+    pub fn to_clean_str(&self) -> Cow<str> {
+        match self {
+            PrimitiveValue::Str(values) => {
+                let new_value = values.trim_matches(|c| c == ' ' || c == '\u{0}');
+                Cow::from(new_value)
+            },
+            PrimitiveValue::Strs(values) => {
+                if values.len() == 1 {
+                    let new_value = values[0].trim_matches(|c| c == ' ' || c == '\u{0}');
+                    Cow::from(new_value)
+                } else {
+                    let mut new_vec = Vec::new();
+                    for i in values{
+                        new_vec.push(i.trim_matches(|c| c == ' ' || c == '\u{0}'));
+                    }
+                    let new_value = new_vec.iter().join("\\").to_owned();
+                    Cow::from(new_value)
```

Comment:
A good rule of thumb is that whenever you find yourself creating a vector and pushing elements into it in a loop, there is likely a more idiomatic way via the iterator API.

```rust
let new_vec: Vec<_> = values.iter()
    .map(|s| s.trim_end_matches(|c| c == ' ' || c == '\0'))
    .collect();
```

In this case however, you don't really need a vector at all, since join works with iterators as well.

```suggestion
                    Cow::Owned(
                        values
                            .iter()
                            .map(|s| s.trim_end_matches(|c| c == ' ' || c == '\0'))
                            .join("\\"),
                    )
```